### PR TITLE
Updating Group 3 project with current code V1

### PR DESCRIPTION
**Pull request by clockian-NAU**

### CODE REVIEW

File: assignmentPR/src/parser.c [hunk 1]
```diff
@@ -12,28 +12,372 @@
...
+ * @detail Ensure the destination str is big enough for the two str
+ *
+ * @param *dest Str that source will be combined into
+ *
+ * @param *source Str that will be put at the end of the dest
+ *
+ * @return Str containing dest and source combined
+ */
+char * strCat(char *dest, char *source){
+    char *copyPtr = dest + strLength(dest);
```
Comment (kww43) [inline assignmentPR/src/parser.c:331]: This is nitpicky but try to keep the same formatting when initializing a pointer. Ex: keep all pointers initialized like: char * ptr;

File: assignmentPR/src/parser.h [hunk 2]
```diff
@@ -11,14 +11,24 @@
...
+int strLength(char *str);
+bool removeLeadingZeros(char *numString);
+bool removeTrailingZeros(char *numString);
+bool removeLeadingSpaces(char *numString);
+bool removeTrailingSpaces(char *numString);
+bool addLeadingZero(char *numString, char *new_string);
+char *getCharacteristic(char *numString, char *newString);
+char *getMantissa(char *numString, char *newString);
+int getMantissaLength(char *numString);
+char * strCat(char *dest, char *source);
```
Comment (kww43) [inline assignmentPR/src/parser.h:30]: Also nitpicky but keep the same formatting as you did for your other function declarations.

File: assignmentPR/src/string_to_int_converter.c [hunk 3]
```diff
@@ -34,15 +35,23 @@
  * @return Bool indicating success or failure
  */
 bool characteristic(char numString[], int *c){
```
Comment (kww43) [inline assignmentPR/src/string_to_int_converter.c:37]: Change c to a more clear variable name so it better explains what it does.

**Review by kww43 — CHANGES_REQUESTED** [review]

**Review by Swhite9478 — APPROVED** [review]
I really like the fact that you guys wrote unit tests! Totally did not think of doing that in our code, but great way to ensure everything works as anticipated! Great work!

**Pull request merged**